Desktop/analysis/analysis.cpp

```diff
@@ -431,59 +434,66 @@ Json::Value& Analysis::_getParentBoundValue(const QVector<JASPControl::ParentKey
 						row[parent.key] = newValue;
 					}
 					parentBoundValues.append(row);
-					result = &(parentBoundValues[parentBoundValues.size() - 1]);
+					parentBoundValue = &(parentBoundValues[parentBoundValues.size() - 1]);
 					found = true;
 				}
 			}
 		}
 	}
 
-	return *result;
+	return result;
 }
 
 bool Analysis::setBoundValue(const std::string &name, const Json::Value &value, const Json::Value &meta, const QVector<JASPControl::ParentKey>& parentKeys)
 {
 	bool found = false;
-	Json::Value &	parentBoundValue	= _getParentBoundValue(parentKeys, found, true),
-					copyPBV				= parentBoundValue; 
+	Json::Value* parentBoundValue	= &_boundValues;
+	QVector<std::string> parents	= _getParentBoundValue(parentKeys, parentBoundValue, found, true);
+	Json::Value copyPBV				= parentBoundValue;
 
-	if (found && parentBoundValue.isObject())
+	if (found && parentBoundValue->isObject())
 	{
-		parentBoundValue[name] = value;
+		(*parentBoundValue)[name] = value;
 
 		if ((meta.isObject() || meta.isArray()) && meta.size() > 0)
-			_boundValues[".meta"][name] = meta;
+		{
+			Json::Value* metaBoundValue = &_boundValues[".meta"];
```

Comment:
This block of code for instance  looks way too complicated. While in fact you just want to pass the `meta` to the parent right?

Reply:
The main issue here is that I need a reference where the meta has to be set, but this reference depends on the parents vector, and in C++, you cannot reassign a reference to another object. That's why I need a pointer.